src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -2352,12 +2352,25 @@ class FunctionSegment(ansi.FunctionSegment):
         ),
         Sequence(
             Sequence(
-                Ref(
-                    "FunctionNameSegment",
-                    exclude=OneOf(
-                        Ref("DatePartFunctionNameSegment"),
-                        Ref("ValuesClauseSegment"),
-                        Ref("ConvertFunctionNameSegment"),
+                OneOf(
+                    Ref(
+                        "FunctionNameSegment",
+                        exclude=OneOf(
+                            Ref("DatePartFunctionNameSegment"),
+                            Ref("ValuesClauseSegment"),
+                            Ref("ConvertFunctionNameSegment"),
+                        ),
+                    ),
+                    Sequence(
+                        Ref.keyword("APPROXIMATE"),
+                        Ref(
+                            "FunctionNameSegment",
+                            exclude=OneOf(
+                                Ref("DatePartFunctionNameSegment"),
+                                Ref("ValuesClauseSegment"),
+                                Ref("ConvertFunctionNameSegment"),
+                            ),
+                        ),
```

Comment:
Hi @greg-finley, that's what I initially tried out, but then it fails parsing an UDF called `approximate`.

But as mentioned in the initial PR comment, there might be a more elegant solution! :sweat_smile:

Reply:
Can we indeed make a UDF with that name? I don't have any Redshift access, so can't test it myself.

Can we add to your test the SQL that would make the UDF and use it?

Reply:
And/or some test that only passes using your code, not the suggestion (or point me to it if one of those SQLs fails under the suggestion)?

Reply:
So in order:
- yes it is possible to create an UDF with that name; it's been tested on a Redshift cluster and validated through AWS documentation list of reserved keywords: https://docs.aws.amazon.com/redshift/latest/dg/r_pg_keywords.html
- Adding the SQL to create the UDF seems out of scope to me when looking at other test files. Esp. since content of the UDF could be anything
- That is the purpose of the last test `select approximate([...]`, which fails with the suggestion. Sorry if that wasn't clear in my previous comment.

Reply:
Gotcha, thanks for your explanation! Sorry for all the questions.